photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/AvroUtils.scala

```diff
@@ -290,33 +289,30 @@ object AvroUtils {
         }
       }
       .distinct(numPartitions)
-      .collect()
-      .toSet
   }
 
   /**
-   * Generate the [[NameAndTermFeatureSetContainer]] from a [[RDD]] of [[GenericRecord]]s.
+   * Generate a [[Map]] of feature section key to [[NameAndTerm]] feature [[RDD]] from a [[RDD]] of [[GenericRecord]]s.
    *
    * @param genericRecords The input [[RDD]] of [[GenericRecord]]s.
    * @param featureSectionKeys The set of feature section keys of interest in the input generic records
-   * @return The generated [[NameAndTermFeatureSetContainer]]
+   * @return The generated [[Map]] of feature section key to [[NameAndTerm]] feature [[RDD]]
    */
-  protected[avro] def readNameAndTermFeatureSetContainerFromGenericRecords(
+  protected[avro] def readNameAndTermFeatureMapFromGenericRecords(
       genericRecords: RDD[GenericRecord],
       featureSectionKeys: Set[String],
-      numPartitions: Int): NameAndTermFeatureSetContainer = {
+      numPartitions: Int): Map[String, RDD[NameAndTerm]] = {
 
-    val nameAndTermFeatureSets = featureSectionKeys
+    featureSectionKeys
       .map { featureSectionKey =>
         (featureSectionKey,
-          AvroUtils.readNameAndTermSetFromGenericRecords(
+          AvroUtils.readNameAndTermsFromGenericRecords(
             genericRecords,
             featureSectionKey,
             numPartitions))
       }
       .toMap
 
-    new NameAndTermFeatureSetContainer(nameAndTermFeatureSets)
   }
```

Comment:
Outermost braces are unnecessary for `readNameAndTermFeatureMapFromGenericRecords`.